packages/ui/src/BlogAppbar.tsx

```diff
@@ -22,16 +22,34 @@ export const BlogAppbar = ({ problem, track }: { problem: Problem; track: Track
 
   const router = useRouter();
 
+  const [prevScrollPos, setPrevScrollPos] = useState(0);
+  const [visible, setVisible] = useState(true);
+  const [scrollingDown, setScrollingDown] = useState(false);
+
+  const handleScroll = () => {
+    const currentScrollPos = window.scrollY;
+    setVisible(prevScrollPos > currentScrollPos || currentScrollPos < 50);
+    setScrollingDown(prevScrollPos < currentScrollPos);
+    setPrevScrollPos(currentScrollPos);
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [prevScrollPos]);
```

Comment:
Can you debounce the call to `handleScroll` ?

Reply:
I have added debounce to handleScroll , you can review it and change the delay in ms as neeed .

Reply:
https://github.com/code100x/daily-code/assets/101008478/e351766e-4a0a-4249-b50b-47c891bef7dd